torchvision/transforms/functional.py

```diff
@@ -713,18 +720,36 @@ def rotate(img, angle, resample=False, expand=False, center=None, fill=0):
         center (2-tuple, optional): Optional center of rotation.
             Origin is the upper left corner.
             Default is the center of the image.
-        fill (3-tuple or int): RGB pixel fill value for area outside the rotated image.
-            If int, it is used for all channels respectively.
+        fill (n-tuple or int or float): Pixel fill value for area outside the rotated
+            image. If int or float, the value is used for all bands respectively.
+            Defaults to 0 for all bands. This option is only available for ``pillow>=5.2.0``.
 
     .. _filters: https://pillow.readthedocs.io/en/latest/handbook/concepts.html#filters
 
     """
+    def verify_fill(fill, num_bands):
+        if PILLOW_VERSION < "5.2.0":
+            msg = ("The option to fill background area of the rotated image, "
+                   "requires pillow>=5.2.0")
+            raise RuntimeError(msg)
+
+        if fill is None:
+            fill = 0
+
+        if isinstance(fill, (int, float)):
```

Comment:
Although the [documentation](https://pillow.readthedocs.io/en/latest/reference/Image.html#PIL.Image.new) states that for single-banded images a `int` or `float` is required, a tuple of length 1 also works fine.

Reply:
This actually doesn't work for images of type `F`.
```python
image = PIL.Image.fromarray(np.zeros((40, 16)))
F.rotate(image, 10, fill=None)
```
gives as error
```python
TypeError                                 Traceback (most recent call last)
<ipython-input-26-1c0cd63212f0> in <module>()
----> 1 torchvision.transforms.functional.rotate(image, 10, fill=None)

~/github/vision/torchvision/transforms/functional.py in rotate(img, angle, resample, expand, center, fill)
    740     opts = parse_fill(fill, len(img.getbands()))
    741
--> 742     return img.rotate(angle, resample, expand, center, **opts)
    743
    744

~/anaconda3/lib/python3.6/site-packages/PIL/Image.py in rotate(self, angle, resample, expand, center, translate, fillcolor)
   1888             w, h = nw, nh
   1889
-> 1890         return self.transform((w, h), AFFINE, matrix, resample, fillcolor=fillcolor)
   1891
   1892     def save(self, fp, format=None, **params):

~/anaconda3/lib/python3.6/site-packages/PIL/Image.py in transform(self, size, method, data, resample, fill, fillcolor)
   2178             raise ValueError("missing method data")
   2179
-> 2180         im = new(self.mode, size, fillcolor)
   2181         if method == MESH:
   2182             # list of quads

~/anaconda3/lib/python3.6/site-packages/PIL/Image.py in new(mode, size, color)
   2348         color = ImageColor.getcolor(color, mode)
   2349
-> 2350     return Image()._new(core.fill(mode, size, color))
   2351
   2352
```

I'm preparing a fix for it